CRM/Contribute/Form/ContributionBase.php

```diff
@@ -567,20 +565,20 @@ public function assignToTemplate() {
     }
 
     //fix for CRM-3767
-    $assignCCInfo = FALSE;
+    $isMonetary = FALSE;
     if ($this->_amount > 0.0) {
-      $assignCCInfo = TRUE;
+      $isMonetary = TRUE;
     }
     elseif (!empty($this->_params['selectMembership'])) {
       $memFee = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_MembershipType', $this->_params['selectMembership'], 'minimum_fee');
       if ($memFee > 0.0) {
-        $assignCCInfo = TRUE;
+        $isMonetary = TRUE;
       }
     }
 
     // The concept of contributeMode is deprecated.
     // The payment processor object can provide info about the fields it shows.
-    if ($assignCCInfo) {
+    if ($isMonetary) {
```

Comment:
I think we do need the `&& $this->_paymentProcessor` here.  Prior to CRM-21134 it had code that did that, and it breaks if you use CiviDiscount with 100% discount as no payment processor will be submitted - there may be other cases too.
Agree with the variable name change.